[debug] [2021-07-15T21:24:53.479Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:24:53.482Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:24:53.482Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:24:53.482Z] Platform:      linux
[debug] [2021-07-15T21:24:53.482Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:24:53.483Z] Time:          Thu Jul 15 2021 23:24:53 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:24:53.483Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:24:53.491Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:24:53.492Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:24:56.536Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:24:57.018Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:25:40.841Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:25:40.843Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:25:40.844Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:25:40.844Z] Platform:      linux
[debug] [2021-07-15T21:25:40.844Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:25:40.845Z] Time:          Thu Jul 15 2021 23:25:40 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:25:40.845Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:25:40.853Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:25:40.854Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:25:43.903Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:25:50.254Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:27:27.793Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:27:27.796Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:27:27.797Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:27:27.797Z] Platform:      linux
[debug] [2021-07-15T21:27:27.797Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:27:27.797Z] Time:          Thu Jul 15 2021 23:27:27 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:27:27.798Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:27:27.806Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:27:27.806Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:27:30.858Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:27:31.230Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:30:19.826Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:30:19.829Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:30:19.829Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:30:19.829Z] Platform:      linux
[debug] [2021-07-15T21:30:19.830Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:30:19.830Z] Time:          Thu Jul 15 2021 23:30:19 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:30:19.830Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:30:19.839Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:30:19.839Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:30:22.884Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:30:23.488Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:31:41.276Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:31:41.279Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:31:41.279Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:31:41.279Z] Platform:      linux
[debug] [2021-07-15T21:31:41.279Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:31:41.280Z] Time:          Thu Jul 15 2021 23:31:41 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:31:41.280Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:31:41.288Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:31:41.288Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:31:44.356Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:31:44.764Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:32:18.439Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:32:18.442Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start --only functions
[debug] [2021-07-15T21:32:18.443Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:32:18.443Z] Platform:      linux
[debug] [2021-07-15T21:32:18.443Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:32:18.444Z] Time:          Thu Jul 15 2021 23:32:18 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:32:18.444Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:32:18.453Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:32:18.453Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions"}}
[debug] [2021-07-15T21:32:21.497Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:32:21.814Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:33:28.159Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:33:28.162Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start --only functions
[debug] [2021-07-15T21:33:28.163Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:33:28.163Z] Platform:      linux
[debug] [2021-07-15T21:33:28.163Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:33:28.163Z] Time:          Thu Jul 15 2021 23:33:28 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:33:28.164Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:33:28.173Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:33:28.173Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions"}}
[debug] [2021-07-15T21:33:31.227Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:33:31.555Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:33:42.247Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:33:42.250Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:33:42.250Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:33:42.251Z] Platform:      linux
[debug] [2021-07-15T21:33:42.251Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:33:42.251Z] Time:          Thu Jul 15 2021 23:33:42 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:33:42.251Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:33:42.260Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:33:42.260Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:33:45.311Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:33:45.628Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:36:01.368Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:36:01.373Z] Command:       /usr/bin/node /usr/bin/firebase functions:shell
[debug] [2021-07-15T21:36:01.373Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:36:01.373Z] Platform:      linux
[debug] [2021-07-15T21:36:01.374Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:36:01.374Z] Time:          Thu Jul 15 2021 23:36:01 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:36:01.375Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:36:01.386Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:36:01.386Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[debug] [2021-07-15T21:36:01.386Z] [iam] checking project clone-6f1af for permissions ["firebase.projects.get"]
[debug] [2021-07-15T21:36:01.389Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/clone-6f1af:testIamPermissions  
 {"permissions":["firebase.projects.get"]}
[debug] [2021-07-15T21:36:02.442Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Thu, 15 Jul 2021 21:36:02 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","server-timing":"gfet4t7; dur=686","alt-svc":"h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2021-07-15T21:36:02.756Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:52:31.764Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:52:31.768Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T21:52:31.769Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:52:31.770Z] Platform:      linux
[debug] [2021-07-15T21:52:31.770Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:52:31.771Z] Time:          Thu Jul 15 2021 23:52:31 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:52:31.771Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:52:31.786Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:52:31.787Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T21:52:34.857Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T21:52:35.381Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T21:56:35.522Z] ----------------------------------------------------------------------
[debug] [2021-07-15T21:56:35.525Z] Command:       /usr/bin/node /usr/bin/firebase functions:shell
[debug] [2021-07-15T21:56:35.525Z] CLI Version:   9.14.0
[debug] [2021-07-15T21:56:35.525Z] Platform:      linux
[debug] [2021-07-15T21:56:35.525Z] Node Version:  v14.16.0
[debug] [2021-07-15T21:56:35.526Z] Time:          Thu Jul 15 2021 23:56:35 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T21:56:35.526Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T21:56:35.534Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T21:56:35.535Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[debug] [2021-07-15T21:56:35.535Z] [iam] checking project clone-6f1af for permissions ["firebase.projects.get"]
[debug] [2021-07-15T21:56:35.538Z] > refreshing access token with scopes: ["email","https://www.googleapis.com/auth/cloud-platform","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","openid"]
[debug] [2021-07-15T21:56:35.538Z] >>> HTTP REQUEST POST https://www.googleapis.com/oauth2/v3/token  
 <request body omitted>
[debug] [2021-07-15T21:56:35.910Z] <<< HTTP RESPONSE 200 {"date":"Thu, 15 Jul 2021 21:56:35 GMT","cache-control":"no-cache, no-store, max-age=0, must-revalidate","pragma":"no-cache","expires":"Mon, 01 Jan 1990 00:00:00 GMT","content-type":"application/json; charset=utf-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","server":"scaffolding on HTTPServer2","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2021-07-15T21:56:36.019Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/clone-6f1af:testIamPermissions  
 {"permissions":["firebase.projects.get"]}
[debug] [2021-07-15T22:05:02.859Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:05:02.862Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:05:02.862Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:05:02.862Z] Platform:      linux
[debug] [2021-07-15T22:05:02.862Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:05:02.863Z] Time:          Fri Jul 16 2021 00:05:02 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:05:02.863Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:05:02.871Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:05:02.871Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:05:05.929Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:05:06.333Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:22:30.242Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:22:30.253Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start --only functions
[debug] [2021-07-15T22:22:30.255Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:22:30.255Z] Platform:      linux
[debug] [2021-07-15T22:22:30.256Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:22:30.293Z] Time:          Fri Jul 16 2021 00:22:30 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:22:30.294Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:22:30.380Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:22:30.382Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions"}}
[debug] [2021-07-15T22:22:33.438Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:22:33.898Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:26:46.544Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:26:46.547Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start --only functions
[debug] [2021-07-15T22:26:46.548Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:26:46.548Z] Platform:      linux
[debug] [2021-07-15T22:26:46.549Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:26:46.549Z] Time:          Fri Jul 16 2021 00:26:46 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:26:46.550Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:26:46.562Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:26:46.562Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions"}}
[debug] [2021-07-15T22:26:49.636Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:26:50.082Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:26:56.979Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:26:56.983Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:26:56.984Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:26:56.984Z] Platform:      linux
[debug] [2021-07-15T22:26:56.984Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:26:56.985Z] Time:          Fri Jul 16 2021 00:26:56 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:26:56.986Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:26:56.994Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:26:56.995Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:27:00.064Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:27:01.436Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:29:01.527Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:29:01.529Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:29:01.530Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:29:01.530Z] Platform:      linux
[debug] [2021-07-15T22:29:01.530Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:29:01.531Z] Time:          Fri Jul 16 2021 00:29:01 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:29:01.531Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:29:01.539Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:29:01.540Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:29:04.586Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:29:08.197Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:35:13.096Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:35:13.099Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:35:13.099Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:35:13.099Z] Platform:      linux
[debug] [2021-07-15T22:35:13.099Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:35:13.100Z] Time:          Fri Jul 16 2021 00:35:13 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:35:13.100Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:35:13.109Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:35:13.109Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:35:16.157Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:35:16.483Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:36:25.125Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:36:25.129Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:36:25.130Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:36:25.130Z] Platform:      linux
[debug] [2021-07-15T22:36:25.130Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:36:25.131Z] Time:          Fri Jul 16 2021 00:36:25 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:36:25.131Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:36:25.143Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:36:25.144Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:36:28.211Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:36:28.598Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2021-07-15T22:36:58.220Z] ----------------------------------------------------------------------
[debug] [2021-07-15T22:36:58.224Z] Command:       /usr/bin/node /usr/bin/firebase emulators:start
[debug] [2021-07-15T22:36:58.225Z] CLI Version:   9.14.0
[debug] [2021-07-15T22:36:58.225Z] Platform:      linux
[debug] [2021-07-15T22:36:58.225Z] Node Version:  v14.16.0
[debug] [2021-07-15T22:36:58.226Z] Time:          Fri Jul 16 2021 00:36:58 GMT+0200 (Central European Summer Time)
[debug] [2021-07-15T22:36:58.226Z] ----------------------------------------------------------------------
[debug] 
[debug] [2021-07-15T22:36:58.238Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2021-07-15T22:36:58.238Z] > authorizing via signed-in user (bmarkoski@gmail.com)
[info] i  emulators: Starting emulators: functions, hosting {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: functions, hosting"}}
[debug] [2021-07-15T22:37:01.304Z] port check error: Error: connect EHOSTUNREACH 192.168.0.102:4400
[info] i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
[debug] [2021-07-15T22:37:01.657Z] Error: Provided host localhost could NOT be bound. Please provide a different host address or hostname
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:174:27
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:473:16
    at iterateeCallback (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:988:17)
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/node_modules/async/dist/async.js:969:16
    at /usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:152:16
    at Server.onError (/usr/lib/node_modules/firebase-tools/node_modules/portfinder/lib/portfinder.js:52:14)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:315:20)
    at Server.EventEmitter.emit (domain.js:467:12)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[error] 
[error] Error: An unexpected error has occurred.
